### ISSUE TITLE
Need refactoring of system settings

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#14539]
### Summary
In the "System Settings" a lot of settings initially. But besides this, there are also settings that are not used in the system at all, but hang in the list.

Those we need refactoring of "System Settings" section:

1) It is necessary to remove outdated system settings.
2) Some settings are not in the right categories, for example:
**publish_default** is located in the "Site" section, but the **default_duplicate_publish_option** setting is located in the "Back-end Manager" section, although, in my opinion, everything related to resources should be in the "Site" section.
**log_snippet_not_found** setting should be in the "Back-end Manager" section.
**default_media_source** and **default_media_source_type** settings should be in the "File System" section etc.
Need to check the categories of system settings.

**Comment by webinmd** [comment]
These settings are not used anywhere in the code.
Most likely this is the legacy of Evo (WebSignup & WebLogin)

- webpwdreminder_message
- websignupemail_message

**Comment by Ruslan-Aleev** [comment]
### In my opinion, these settings do not work (maybe i'm wrong):

**~~Authentication and Security:~~** (https://github.com/modxcms/revolution/pull/14841)
- ~~udperms_allowroot (found only in "System Settings")~~
- ~~webpwdreminder_message (found only in "System Settings")~~

**~~Rich-Text Editor:~~** (https://github.com/modxcms/revolution/pull/14843)
- ~~editor_css_path (found only in "System Settings")~~
- ~~editor_css_selectors (found only in "System Settings")~~

**~~File System:~~** (https://github.com/modxcms/revolution/pull/14886)
- ~~filemanager_path (although it is indicated as deprecated, the setting is found in the files)~~
- ~~filemanager_path_relative (found only in "System Settings")~~
- ~~filemanager_url (although it is indicated as deprecated, the setting is found in the files)~~
- ~~filemanager_url_relative (found only in "System Settings")~~
- ~~rb_base_dir (?)~~
- ~~rb_base_url (?)~~
- ~~strip_image_paths (found only in "System Settings")~~
- ~~use_browser (found only in "System Settings")~~

**~~Lexicon and Language:~~** (https://github.com/modxcms/revolution/pull/14848)
- ~~fe_editor_lang (found only in "System Settings")~~

**~~Back-end Manager:~~** (https://github.com/modxcms/revolution/pull/14868)
- ~~manager_js_cache_file_locking (found only in "System Settings")~~
- ~~manager_js_cache_max_age (found only in "System Settings")~~
- ~~manager_js_document_root (found only in "System Settings")~~
- inline_help (?)

**~~System and Server:~~** (https://github.com/modxcms/revolution/pull/14877)
- ~~resolve_hostnames (found only in "System Settings")~~
- ~~server_protocol (https://github.com/modxcms/revolution/issues/13269)~~

**It is better to check all system settings, they can be even called in the core files, but not used in core functions.**

---

### In my opinion, these settings need to be moved to another category:

**Perhaps it is worth moving to "Authentication and Security" section:**
- tv_eval (?)

**To "Caching" section:**
- manager_js_cache_max_age
- manager_js_cache_file_locking
- parser_recurse_uncacheable (?)

**To "File System" section:**
- modx_browser_default_sort
- default_media_source
- default_media_source_type
- modx_browser_tree_hide_files
- modx_browser_tree_hide_tooltips
- modx_browser_default_viewmode

**To "Friendly URL" section:**
- link_tag_scheme (?)

**To "Back-end Manager" section:**
- feed_modx_news_enabled
- feed_modx_news
- feed_modx_security_enabled
- feed_modx_security

**To "Site" section:**
- preserve_menuindex
- default_duplicate_publish_option
- lock_ttl
- use_context_resource_table
- tvs_below_content

**To "System and Server" section:**
- log_snippet_not_found

**It is better to move these settings from the "System settings" in the "User" section in the form of checkboxes:**
- manager_use_fullname
- enable_gravatar

**Comment by JoshuaLuckers** [comment]
Why should the setting `enable_gravatar` be removed?

**Comment by Ruslan-Aleev** [comment]
@JoshuaLuckers I meant that it is better to move these settings (manager_use_fullname, enable_gravatar) to the "User" section in the form of checkboxes, don't just remove (misprint).

**Comment by JoshuaLuckers** [comment]
> @JoshuaLuckers I meant that it is better to move these settings (manager_use_fullname, enable_gravatar) to the "User" section in the form of checkboxes, don't just remove (misprint).

Thanks for clearing that up! 😄

**Comment by JoshuaLuckers** [comment]
The system setting `server_protocol` should be removed.

**Comment by Ruslan-Aleev** [comment]
@JoshuaLuckers In a good way, all the settings should be checked, there are a lot of old non-working settings. I described some above.

**Comment by Jako** [comment]
The `static_elements_automate_…` settings are in use:
https://github.com/modxcms/revolution/blob/3.x/core/src/Revolution/Processors/Element/Remove.php#L80

**Comment by Ruslan-Aleev** [comment]
@Jako Yes, thank you. I wanted to clarify this later, I searched by the direct key and not find this settings (`static_elements_automate_...`).
Can you check other settings too, when you have time?

**Comment by JoshuaLuckers** [comment]
> **Authentication and Security:** (#14841)
> 
> * udperms_allowroot (found only in "System Settings")
> * webpwdreminder_message (found only in "System Settings")

These settings have been deleted.
